lib/akismet.js

```diff
@@ -37,16 +41,13 @@ var Akismet = function(options) {
       blog : this.blog
     })
     .set('User-Agent', this.userAgent)
-    .end(function(err, res) {
-      if (err) cb(err, null);
-      else if (res.text == 'valid') cb(null, true);
-      else if (res.header['x-akismet-debug-help']) 
-        cb(new Error(res.header['x-akismet-debug-help']), false);
-      else if (res.text == 'invalid')
-        cb(new Error('Invalid API key'), false);
-      else 
-        cb(new Error(res.text), false);
-    });
+    .endAsync()
+    .then(function(res) {
+      if (res.text == 'valid') return true;
+      if (res.text == 'invalid') return false;
+      throw new Error(res.text);
```

Comment:
Did you leave out the "new Error(res.header['x-akismet-debug-help']" error case on purpose?